src/Controller/Admin/UserController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Admin;
+
+use App\Entity\User;
+use App\Form\UserUpdateFormType;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class UserController extends AbstractController
+{
+    private UserRepository $userRepository;
+
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(UserRepository $userRepository, EntityManagerInterface $entityManager)
+    {
+        $this->userRepository = $userRepository;
+        $this->entityManager = $entityManager;
+    }
+
+
+    /**
+     * @Route(path="/users", methods={"GET"}, name="list_users")
+     */
+    public function listAllUsers(): Response
+    {
+        $users = $this->userRepository->findAll();
+
+        return $this->render('admin/listUsers.html.twig', [
+            'users' => $users,
+        ]);
+    }
+
+    /**
+     * @Route(path="/users/{id}", methods={"GET", "POST"}, name="update_user")
+     */
+    public function updateUserAction(int $id, Request $request): Response
+    {
+        $userEntity = new User();
+
+        $form = $this->createForm(UserUpdateFormType::class, $userEntity);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $user = $this->userRepository->findOneBy(['id' => $id]);
+
+            $firstName = $form->get('firstName')->getData();
+            $lastName = $form->get('lastName')->getData();
+            $email = $form->get('email')->getData();
+            $phoneNumber = $form->get('phoneNumber')->getData();
+
+            $user->firstName = $firstName;
+            $user->lastName = $lastName;
+            $user->email = $email;
+            $user->phoneNumber = $phoneNumber;
+
```

Comment:
you can do this more easily by calling `getData` just once, the user is already loaded in the entity